pkg/operator/configobservation/apiserver/observe_shutdown_delay.go

```diff
@@ -0,0 +1,64 @@
+package apiserver
+
+import (
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var shutdownDelayDurationPath = []string{"apiServerArguments", "shutdown-delay-duration"}
+
+// ObserveShutdownDelayDuration allows for overwriting shutdown-delay-duration value.
+// It exists because the time needed for an LB to notice and remove unhealthy instances might vary by platform.
+func ObserveShutdownDelayDuration(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	var currentShutdownDelayDuration string
+	var errs []error
+
+	// read the current value
+	currentShutdownDelaySlice, _, err := unstructured.NestedStringSlice(existingConfig, shutdownDelayDurationPath...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to extract shutdown delay duration from the existing config: %v", err))
+	}
+
+	if len(currentShutdownDelaySlice) > 0 {
+		currentShutdownDelayDuration = currentShutdownDelaySlice[0]
+	}
+
+	// read the observed value
+	var observedShutdownDelayDuration string
+	listers := genericListers.(configobservation.Listers)
+	infra, err := listers.InfrastructureLister().Get("cluster")
+	if err != nil && !apierrors.IsNotFound(err) {
+		// without the infrastructure object we are not able to determine the type of platform we are running on
+		return configobserver.Pruned(existingConfig, shutdownDelayDurationPath), append(errs, err)
```

Comment:
you don't like my pretty defer based pruning call?

Reply:
I do :)

Added.